src/test/java/org/opensearch/neuralsearch/query/HybridQueryIT.java

```diff
@@ -202,23 +215,104 @@ public void testNestedQuery_whenHybridQueryIsWrappedIntoOtherQuery_thenSuccess()
+        );
+
+        org.hamcrest.MatcherAssert.assertThat(
+            exceptionNoNestedTypes.getMessage(),
+            allOf(
+                containsString("hybrid query must be a top level query and cannot be wrapped into other queries"),
+                containsString("illegal_argument_exception")
+            )
+        );
+
+        initializeIndexIfNotExist(TEST_MULTI_DOC_INDEX_WITH_NESTED_TYPE_NAME_ONE_SHARD);
+
+        ResponseException exceptionQWithNestedTypes = expectThrows(
+            ResponseException.class,
+            () -> search(
+                TEST_MULTI_DOC_INDEX_WITH_NESTED_TYPE_NAME_ONE_SHARD,
+                boolQueryBuilder,
+                null,
+                10,
+                Map.of("search_pipeline", SEARCH_PIPELINE)
+            )
+        );
+
+        org.hamcrest.MatcherAssert.assertThat(
+            exceptionQWithNestedTypes.getMessage(),
+            allOf(
+                containsString("hybrid query must be a top level query and cannot be wrapped into other queries"),
+                containsString("illegal_argument_exception")
+            )
+        );
+    }
+
+    @SneakyThrows
+    public void testIndexWithNestedFields_whenHybridQuery_thenSuccess() {
+        initializeIndexIfNotExist(TEST_MULTI_DOC_INDEX_WITH_NESTED_TYPE_NAME_ONE_SHARD);
+
+        TermQueryBuilder termQueryBuilder = QueryBuilders.termQuery(TEST_TEXT_FIELD_NAME_1, TEST_QUERY_TEXT);
+        TermQueryBuilder termQuery2Builder = QueryBuilders.termQuery(TEST_TEXT_FIELD_NAME_1, TEST_QUERY_TEXT2);
+        HybridQueryBuilder hybridQueryBuilderOnlyTerm = new HybridQueryBuilder();
+        hybridQueryBuilderOnlyTerm.add(termQueryBuilder);
+        hybridQueryBuilderOnlyTerm.add(termQuery2Builder);
+
         Map<String, Object> searchResponseAsMap = search(
-            TEST_MULTI_DOC_INDEX_NAME_ONE_SHARD,
-            boolQueryBuilder,
+            TEST_MULTI_DOC_INDEX_WITH_NESTED_TYPE_NAME_ONE_SHARD,
+            hybridQueryBuilderOnlyTerm,
             null,
             10,
             Map.of("search_pipeline", SEARCH_PIPELINE)
         );
 
-        assertTrue(getHitCount(searchResponseAsMap) > 0);
+        assertEquals(0, getHitCount(searchResponseAsMap));
         assertTrue(getMaxScore(searchResponseAsMap).isPresent());
-        assertTrue(getMaxScore(searchResponseAsMap).get() > 0.0f);
+        assertEquals(0.0f, getMaxScore(searchResponseAsMap).get(), DELTA_FOR_SCORE_ASSERTION);
 
         Map<String, Object> total = getTotalHits(searchResponseAsMap);
         assertNotNull(total.get("value"));
-        assertTrue((int) total.get("value") > 0);
+        assertEquals(0, total.get("value"));
+        assertNotNull(total.get("relation"));
+        assertEquals(RELATION_EQUAL_TO, total.get("relation"));
     }
 
+    @SneakyThrows
+    public void testIndexWithNestedFields_whenHybridQueryIncludesNested_thenSuccess() {
```

Comment:
for these integration tests can we validate what should be the expected score. This will ensure that normalization is happening.

Reply:
ack